### ISSUE TITLE
Effect of learning rate scaling

### ISSUE THREAD

**Issue by leila000** [AlexeyAB/darknet#4149]
**- scales=.1,.1,.1 - if policy=steps - f.e. if steps=8000,9000,12000, scales=.1,.1,.1 and the current iteration number is 10000 then current_learning_rate = learning_rate * scales[0] * scales[1] = 0.001 * 0.1 * 0.1 = 0.00001**

I saw this and I got one question
what is a difference between setting steps=4000,5000 and steps =40000,5000?
I mean, what effect does these steps bring? How do I know which steps are proper to my own data?

Thanks for answering my question :D

**Comment by mzeimet** [comment]
[steps recommendation](https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects)

Learning rate decreases at given step counts by given factor. It assumes you're not far from the best loss and therefore decrease your "search-speed" to search with more precision. Like traveling by car fast on a highway, slower on a normal street and slowest when looking for a specific house number.

**Comment by AlexeyAB** [comment]
`learning_rate=0.001 steps=8000,9000,12000, scales=.1,.1,.1`

It means that 
* at iterations 0 - 8000, the learning rate will be 0.001
* at iterations 8000 - 9000 , the learning rate will be 0.0001
* at iterations 9000 - 12000 , the learning rate will be 0.00001
* at iterations 12000 - ..., the learning rate will be 0.000001

More: https://github.com/AlexeyAB/darknet/wiki/CFG-Parameters-in-the-%5Bnet%5D-section

**Comment by leila000** [comment]
@AlexeyAB @mzeimet  Both of you thanks a lot ! 
I know the role and definition of it. I was just curious how people set their learning rate and steps. But I guess repeating experiment is obviously the way for finding the best parameters :D

**Comment by hooman67** [comment]
@AlexeyAB  I set my learning rate to 0.001 in the config. But the learning rate printed in the training logs is 0.002 after the burn in. Any ideas why that is? I've never heard of learning rate being set to higher values than specified in the config.

**Comment by AlexeyAB** [comment]
@hooman67 
Show your cfg-file.
What code do you use?
